### DISCUSSION TITLE
Doubt regarding library (libc-bin) detected against nvd CVE cpe (glibc)

### DISCUSSION THREAD

**Discussion by nimish-salve** [aquasecurity/trivy#4187] [Q&A] [answered]
[CVE-2021-38604](https://nvd.nist.gov/vuln/detail/CVE-2021-38604) affects CPE `cpe:2.3:a:gnu:glibc:*:*:*:*:*:*:*:*`. However, trivy detects library `libc-bin` affected by vulnerability CVE-2021-38604. 

As per my understanding (very limited), it should show glibc affected by CVE-2021-38604. There are many similar detections where cpe is different but the library detected by trivy is different. Is there any way/flag in Trivy which gives a cpe name or both as a library?

**Comment by nimish-salve** [comment]
@DmitriyLewen I see a similar thread you have answered. https://github.com/aquasecurity/trivy/issues/4151. Do you care to help me with this?

In the case of https://github.com/aquasecurity/trivy/issues/4151 too, CPE is OpenSSL, but Trivy has detected libssl1.1. 
I want to understand why and how it gets libssl?

Is there any option to retrieve openssl as well as libssl1.1?

**Comment by nimish-salve** [comment]
Can I get any update on this please?

**Comment by DmitriyLewen** [comment, marked as answer]
Hello @nimish-salve 
Thanks for your interest to Trivy!

As you correctly noted  - there is cases, when package name and source name have different names.
[Ubuntu CVE tracker](https://ubuntu.com/security/cves) uses source name, but `apt` shows package name. 
So that you can find package in `apt` we use package name in table report.

> Is there any way/flag in Trivy which gives a cpe name or both as a library?

You can use `--format json --list-all-pkgs` flags to see both names.
e.g.:
```bash
➜  trivy -d image -f json --list-all-pkgs ubuntu | grep '"ID": "libc-bin@2.35-0ubuntu3.1"' -A 7
...
          "ID": "libc-bin@2.35-0ubuntu3.1",
          "Name": "libc-bin",
          "Version": "2.35",
          "Release": "0ubuntu3.1",
          "Arch": "amd64",
          "SrcName": "glibc",
          "SrcVersion": "2.35",
          "SrcRelease": "0ubuntu3.1",
```

**Comment by nimish-salve** [comment]
Thanks for the help @DmitriyLewen.

I have observed in a few cases, SrcName is empty. I believe that is because a particular package is a standalone package or does not have any vulnerability?

**Reply by DmitriyLewen** [reply]
[Language packages](https://aquasecurity.github.io/trivy/v0.41/docs/scanner/vulnerability/language/) don't have `SrcName`. You may have seen these packages.
